Set default entity.type value to User and Host entity store ingest pipelines

opauloh opened the pull request:
## Summary

It closes #219821

This PR adds default `entity.type` value to the User and Host entity store ingest pipelines:

**Default value for User Entity Store:**
`entity.type` = "Identity"

**Default value for Host Entity Store:**
`entity.type` = "Host"


### Entity Store changes:

- Configured the existing `pipeline` parameter to work with either array or custom function as per it's definition. This allows for setting custom processors on the respective `latest@platform` Pipeline
- Added unit tests to the `ingest_pipeline.ts` file


## Screenshot


Ingest pipeline (security_user_default-latest@platform):

<img width="2435" alt="image" src="https://github.com/user-attachments/assets/b8c1c16f-41f5-41fb-86f8-1c387c112422" />

Document:

<img width="2019" alt="image" src="https://github.com/user-attachments/assets/e083aedb-dbe9-4552-a94b-4bb553a3de6d" />



<img width="1220" alt="image" src="https://github.com/user-attachments/assets/93e26ae8-7df5-4ae2-8ff3-ceb0ed0c639c" />

opauloh committed:
Add entity.type Field to User and Host Entity Stores

opauloh committed:
Add entity.type Field to User and Host Entity Stores and pipeline setting capability

elasticmachine commented:
## :hourglass_flowing_sand: Build in-progress, with failures
* [Buildkite Build](https://buildkite.com/elastic/kibana-pull-request/builds/307698)
* Commit: e140f58fe3b631f3e16d4a08e4c7b644948a2cc2

### Failed CI Steps
* [Check Types](https://buildkite.com/elastic/kibana-pull-request/builds/307698#01976327-41d9-42b8-8e2a-53eef2eaa53a)

<!--buildkite-pr-comment-kibana-pull-request
{"builds":[{"buildStatus":{"state":"failing","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/307698","number":307698,"commit":"e140f58fe3b631f3e16d4a08e4c7b644948a2cc2"}],"number":307698}
buildkite-pr-comment-->

hop-dev commented:
@opauloh do we have a plan for how this field will be added to existing customers entity engines?

opauloh committed:
Updating logic to use entity store pipeline instead of the default pipeline

opauloh committed:
revert platform folder changes

opauloh committed:
removing previous logic

opauloh commented:
> @opauloh do we have a plan for how this field will be added to existing customers entity engines?

Not really, existing customers would need to go to the Entity Store page to disable and enable the entity store in order to update the pipeline. (Which wouldn't lead to loss of data, but would re-process the transform lookback time with the new field, older data would remain without type), as the example below:

<img width="1913" alt="image" src="https://github.com/user-attachments/assets/0250e57d-0ead-4aec-8e8f-abfafdf4cd27" />

Maybe we can think about something more sophisticated to identify ingest pipeline changes on the fly that apply to new data, but I don't think we should focus much on updating data that existed previously.

opauloh committed:
adding unit tests to cover most of ingest_pipeline tests

opauloh commented:
Update: initially, I added the processor logic into the Entity Manager pipeline directly, but after double checking, it 
makes sense to leverage the `latest@platform` (security_[ENTITY_STORE]_default-latest@platform) pipeline with all logic in the Entity Store rather than on the Entity Manager.

opauloh committed:
Merge branch 'main' into entity-store/user-host-entity-type

elasticmachine commented:
## :green_heart: Build Succeeded
* [Buildkite Build](https://buildkite.com/elastic/kibana-pull-request/builds/308939)
* Commit: db0f6c0c044fc4c965ccb1a2e5569047f3cfbdf4

### Metrics [[docs](https://www.elastic.co/guide/en/kibana/master/ci-metrics.html)]
<details><summary>Unknown metric groups</summary>

#### ESLint disabled line counts
| id | [before](https://github.com/elastic/kibana/commit/b08d2284a16a2c4cb3aa969fe850a7472903da9b) | [after](https://github.com/elastic/kibana/commit/db0f6c0c044fc4c965ccb1a2e5569047f3cfbdf4) | diff |
| --- | --- | --- | --- |
| `securitySolution` | 609 | 613 | +4 |

#### Total ESLint disabled count
| id | [before](https://github.com/elastic/kibana/commit/b08d2284a16a2c4cb3aa969fe850a7472903da9b) | [after](https://github.com/elastic/kibana/commit/db0f6c0c044fc4c965ccb1a2e5569047f3cfbdf4) | diff |
| --- | --- | --- | --- |
| `securitySolution` | 703 | 707 | +4 |


</details>


### History
* :broken_heart: [Build #308171](https://buildkite.com/elastic/kibana-pull-request/builds/308171) failed 650c11dd906615f73236e946141c09802d01d75d
* :green_heart: [Build #308154](https://buildkite.com/elastic/kibana-pull-request/builds/308154) succeeded 2c6619f6f09adfd96dfdc73d3cde6d85b9344ad6
* :broken_heart: [Build #307698](https://buildkite.com/elastic/kibana-pull-request/builds/307698) failed e140f58fe3b631f3e16d4a08e4c7b644948a2cc2

<!--buildkite-pr-comment-kibana-pull-request
{"builds":[{"buildStatus":{"state":"passed","success":true,"hasRetries":true,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/308939","number":308939,"commit":"db0f6c0c044fc4c965ccb1a2e5569047f3cfbdf4"},{"buildStatus":{"state":"failed","success":false,"hasRetries":true,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/308171","number":308171,"commit":"650c11dd906615f73236e946141c09802d01d75d"},{"buildStatus":{"state":"passed","success":true,"hasRetries":true,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/308154","number":308154,"commit":"2c6619f6f09adfd96dfdc73d3cde6d85b9344ad6"},{"buildStatus":{"state":"failed","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/307698","number":307698,"commit":"e140f58fe3b631f3e16d4a08e4c7b644948a2cc2"}],"number":308939}
buildkite-pr-comment-->

opauloh committed:
Merge branch 'main' into entity-store/user-host-entity-type

elasticmachine commented:
## :green_heart: Build Succeeded
* [Buildkite Build](https://buildkite.com/elastic/kibana-pull-request/builds/311590)
* Commit: 127e532a566b14e25564e1754268066fa679664d

### Metrics [[docs](https://www.elastic.co/guide/en/kibana/master/ci-metrics.html)]
<details><summary>Unknown metric groups</summary>

#### ESLint disabled line counts
| id | [before](https://github.com/elastic/kibana/commit/b1d4ec4b3a5a4c98b12b349b1b46367056c8cd75) | [after](https://github.com/elastic/kibana/commit/127e532a566b14e25564e1754268066fa679664d) | diff |
| --- | --- | --- | --- |
| `securitySolution` | 621 | 625 | +4 |

#### Total ESLint disabled count
| id | [before](https://github.com/elastic/kibana/commit/b1d4ec4b3a5a4c98b12b349b1b46367056c8cd75) | [after](https://github.com/elastic/kibana/commit/127e532a566b14e25564e1754268066fa679664d) | diff |
| --- | --- | --- | --- |
| `securitySolution` | 719 | 723 | +4 |


</details>


### History
* :green_heart: [Build #308939](https://buildkite.com/elastic/kibana-pull-request/builds/308939) succeeded db0f6c0c044fc4c965ccb1a2e5569047f3cfbdf4
* :broken_heart: [Build #308171](https://buildkite.com/elastic/kibana-pull-request/builds/308171) failed 650c11dd906615f73236e946141c09802d01d75d
* :green_heart: [Build #308154](https://buildkite.com/elastic/kibana-pull-request/builds/308154) succeeded 2c6619f6f09adfd96dfdc73d3cde6d85b9344ad6
* :broken_heart: [Build #307698](https://buildkite.com/elastic/kibana-pull-request/builds/307698) failed e140f58fe3b631f3e16d4a08e4c7b644948a2cc2

<!--buildkite-pr-comment-kibana-pull-request
{"builds":[{"buildStatus":{"state":"passed","success":true,"hasRetries":true,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/311590","number":311590,"commit":"127e532a566b14e25564e1754268066fa679664d"},{"buildStatus":{"state":"passed","success":true,"hasRetries":true,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/308939","number":308939,"commit":"db0f6c0c044fc4c965ccb1a2e5569047f3cfbdf4"},{"buildStatus":{"state":"failed","success":false,"hasRetries":true,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/308171","number":308171,"commit":"650c11dd906615f73236e946141c09802d01d75d"},{"buildStatus":{"state":"passed","success":true,"hasRetries":true,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/308154","number":308154,"commit":"2c6619f6f09adfd96dfdc73d3cde6d85b9344ad6"},{"buildStatus":{"state":"failed","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/307698","number":307698,"commit":"e140f58fe3b631f3e16d4a08e4c7b644948a2cc2"}],"number":311590}
buildkite-pr-comment-->

opauloh commented:
@hop-dev, as discussed at the previous Entity Store meeting, I created [this follow-up ticket](https://github.com/elastic/security-team/issues/12877) to investigate how to introduce new fields on Environments with an existing installation of the Entity Store.

cc @romulets

opauloh committed:
Merge branch 'main' into entity-store/user-host-entity-type

opauloh commented:
Reopening based on the discussion in [this issue](https://github.com/elastic/security-team/issues/12877#issuecomment-3100801035), cc @romulets

elasticmachine commented:
## :broken_heart: Build Failed
* [Buildkite Build](https://buildkite.com/elastic/kibana-pull-request/builds/321633)
* Commit: f16c3d3ff580ed1c1067354f76ed6030c094597e

### Failed CI Steps
* [Check Types](https://buildkite.com/elastic/kibana-pull-request/builds/321633#0198304e-a08c-4b04-90e2-1c2c3985eba0)

### History
* :green_heart: [Build #311590](https://buildkite.com/elastic/kibana-pull-request/builds/311590) succeeded 127e532a566b14e25564e1754268066fa679664d
* :green_heart: [Build #308939](https://buildkite.com/elastic/kibana-pull-request/builds/308939) succeeded db0f6c0c044fc4c965ccb1a2e5569047f3cfbdf4
* :broken_heart: [Build #308171](https://buildkite.com/elastic/kibana-pull-request/builds/308171) failed 650c11dd906615f73236e946141c09802d01d75d
* :green_heart: [Build #308154](https://buildkite.com/elastic/kibana-pull-request/builds/308154) succeeded 2c6619f6f09adfd96dfdc73d3cde6d85b9344ad6
* :broken_heart: [Build #307698](https://buildkite.com/elastic/kibana-pull-request/builds/307698) failed e140f58fe3b631f3e16d4a08e4c7b644948a2cc2

<!--buildkite-pr-comment-kibana-pull-request
{"builds":[{"buildStatus":{"state":"failed","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/321633","number":321633,"commit":"f16c3d3ff580ed1c1067354f76ed6030c094597e"},{"buildStatus":{"state":"passed","success":true,"hasRetries":true,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/311590","number":311590,"commit":"127e532a566b14e25564e1754268066fa679664d"},{"buildStatus":{"state":"passed","success":true,"hasRetries":true,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/308939","number":308939,"commit":"db0f6c0c044fc4c965ccb1a2e5569047f3cfbdf4"},{"buildStatus":{"state":"failed","success":false,"hasRetries":true,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/308171","number":308171,"commit":"650c11dd906615f73236e946141c09802d01d75d"},{"buildStatus":{"state":"passed","success":true,"hasRetries":true,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/308154","number":308154,"commit":"2c6619f6f09adfd96dfdc73d3cde6d85b9344ad6"},{"buildStatus":{"state":"failed","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/307698","number":307698,"commit":"e140f58fe3b631f3e16d4a08e4c7b644948a2cc2"}],"number":321633}
buildkite-pr-comment-->

opauloh committed:
Merge branch 'main' into entity-store/user-host-entity-type

opauloh committed:
Bumps entity definition versions

Increments the entity definition versions for host and user entities.

This change reflects updates to the entity definitions, ensuring that the system uses the latest definitions.

elasticmachine commented:
## :hourglass_flowing_sand: Build in-progress, with failures
* [Buildkite Build](https://buildkite.com/elastic/kibana-pull-request/builds/322135)
* Commit: ed5e2facebd4b36413173cedad210bb3690bc8bc

### Failed CI Steps
* [Check Types](https://buildkite.com/elastic/kibana-pull-request/builds/322135#019834e0-fd70-41fc-b969-b4515a6b022b)

### History
* :broken_heart: [Build #322128](https://buildkite.com/elastic/kibana-pull-request/builds/322128) failed 7c4d41b3686ed7300dea9c7ab58ac3f80f2d3371
* :broken_heart: [Build #321633](https://buildkite.com/elastic/kibana-pull-request/builds/321633) failed f16c3d3ff580ed1c1067354f76ed6030c094597e
* :green_heart: [Build #311590](https://buildkite.com/elastic/kibana-pull-request/builds/311590) succeeded 127e532a566b14e25564e1754268066fa679664d
* :green_heart: [Build #308939](https://buildkite.com/elastic/kibana-pull-request/builds/308939) succeeded db0f6c0c044fc4c965ccb1a2e5569047f3cfbdf4
* :broken_heart: [Build #308171](https://buildkite.com/elastic/kibana-pull-request/builds/308171) failed 650c11dd906615f73236e946141c09802d01d75d
* :green_heart: [Build #308154](https://buildkite.com/elastic/kibana-pull-request/builds/308154) succeeded 2c6619f6f09adfd96dfdc73d3cde6d85b9344ad6

<!--buildkite-pr-comment-kibana-pull-request
{"builds":[{"buildStatus":{"state":"failing","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/322135","number":322135,"commit":"ed5e2facebd4b36413173cedad210bb3690bc8bc"},{"buildStatus":{"state":"failed","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/322128","number":322128,"commit":"7c4d41b3686ed7300dea9c7ab58ac3f80f2d3371"},{"buildStatus":{"state":"failed","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/321633","number":321633,"commit":"f16c3d3ff580ed1c1067354f76ed6030c094597e"},{"buildStatus":{"state":"passed","success":true,"hasRetries":true,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/311590","number":311590,"commit":"127e532a566b14e25564e1754268066fa679664d"},{"buildStatus":{"state":"passed","success":true,"hasRetries":true,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/308939","number":308939,"commit":"db0f6c0c044fc4c965ccb1a2e5569047f3cfbdf4"},{"buildStatus":{"state":"failed","success":false,"hasRetries":true,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/308171","number":308171,"commit":"650c11dd906615f73236e946141c09802d01d75d"}],"number":322135}
buildkite-pr-comment-->

opauloh committed:
Merge branch 'main' into entity-store/user-host-entity-type

elasticmachine commented:
## :broken_heart: Build Failed
* [Buildkite Build](https://buildkite.com/elastic/kibana-pull-request/builds/323091)
* Commit: 3ca3eb65b6bc8770fdb8a57f6e8da8b3786a28bf

### Failed CI Steps
* [Check Types](https://buildkite.com/elastic/kibana-pull-request/builds/323091#01983e9b-71be-4471-a342-54f991bac594)

### History
* :broken_heart: [Build #322135](https://buildkite.com/elastic/kibana-pull-request/builds/322135) failed ed5e2facebd4b36413173cedad210bb3690bc8bc
* :broken_heart: [Build #322128](https://buildkite.com/elastic/kibana-pull-request/builds/322128) failed 7c4d41b3686ed7300dea9c7ab58ac3f80f2d3371
* :broken_heart: [Build #321633](https://buildkite.com/elastic/kibana-pull-request/builds/321633) failed f16c3d3ff580ed1c1067354f76ed6030c094597e
* :green_heart: [Build #311590](https://buildkite.com/elastic/kibana-pull-request/builds/311590) succeeded 127e532a566b14e25564e1754268066fa679664d
* :green_heart: [Build #308939](https://buildkite.com/elastic/kibana-pull-request/builds/308939) succeeded db0f6c0c044fc4c965ccb1a2e5569047f3cfbdf4

<!--buildkite-pr-comment-kibana-pull-request
{"builds":[{"buildStatus":{"state":"failed","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/323091","number":323091,"commit":"3ca3eb65b6bc8770fdb8a57f6e8da8b3786a28bf"},{"buildStatus":{"state":"failed","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/322135","number":322135,"commit":"ed5e2facebd4b36413173cedad210bb3690bc8bc"},{"buildStatus":{"state":"failed","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/322128","number":322128,"commit":"7c4d41b3686ed7300dea9c7ab58ac3f80f2d3371"},{"buildStatus":{"state":"failed","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/321633","number":321633,"commit":"f16c3d3ff580ed1c1067354f76ed6030c094597e"},{"buildStatus":{"state":"passed","success":true,"hasRetries":true,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/311590","number":311590,"commit":"127e532a566b14e25564e1754268066fa679664d"},{"buildStatus":{"state":"passed","success":true,"hasRetries":true,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/308939","number":308939,"commit":"db0f6c0c044fc4c965ccb1a2e5569047f3cfbdf4"}],"number":323091}
buildkite-pr-comment-->

opauloh committed:
Adds missing identity field mapping

Ensures that identity field mapping is present in the schema options.
This prevents potential issues related to missing configurations.

romulets commented on x-pack/solutions/security/plugins/security_solution/server/lib/entity_analytics/entity_store/elasticsearch_assets/ingest_pipeline.test.ts:
```diff
@@ -0,0 +1,294 @@
+/*
```
Awesome addition 🤩

romulets commented on x-pack/solutions/security/plugins/security_solution/server/lib/entity_analytics/entity_store/entity_definitions/entity_descriptions/host.ts:
```diff
@@ -46,4 +46,12 @@ export const hostEntityEngineDescription: EntityDescription = {
     collect({ source: 'host.architecture' }),
     ...getCommonFieldDescriptions('host'),
   ],
+  customIngestProcessors: [
```
This comment goes for user and host.

Right now, there is no way to express user type and host type. To simplify at this moment, I agree with hardcoding Host and Identity always.

But the near future once we have the nested `entity` type under `host.*`  and `user*` I would have this processor with `override: false` so we don't override any manual type used.

So this is mostly a reminder for us @uri-weisman , this will need to be updated.

romulets commented on x-pack/platform/plugins/shared/entity_manager/server/lib/entities/ingest_pipeline/generate_latest_processors.ts:
```diff
@@ -6,6 +6,7 @@
  */
 
 import { EntityDefinition, ENTITY_SCHEMA_VERSION_V1, MetadataField } from '@kbn/entities-schema';
+import { IngestProcessorContainer } from '@elastic/elasticsearch/lib/api/types';
```
nice house keeping in this file :)

romulets commented on x-pack/solutions/security/plugins/security_solution/server/lib/entity_analytics/entity_store/installation/types.ts:
```diff
@@ -57,13 +57,19 @@ export interface EntityEngineInstallationDescriptor {
 
+  /**
+   * Custom processors to apply to the entity data.
+   * It's an array of processors which get appended to the platform pipeline of the defined entity.
```
I wonder if we need this new customIngestProcessors or just make `pipelines` do what it claims it does. 

In theory we could export the `pipeline` [here](https://github.com/elastic/kibana/blob/main/x-pack/solutions/security/plugins/security_solution/server/lib/entity_analytics/entity_store/entity_definitions/types.ts#L16-L28) and define it in the definitions.

Then on this [convoluted logic](https://github.com/elastic/kibana/blob/main/x-pack/solutions/security/plugins/security_solution/server/lib/entity_analytics/entity_store/elasticsearch_assets/ingest_pipeline.ts#L117-L119), append the processors of the pipeline case there are any (right now we support only function for some reason)

opauloh replied:
Really good point @romulets, I was initially confused about how the `pipeline` parameter should work, but after reviewing it you were right, the `pipeline` parameter should already handle that however, it was working only with function.

I updated it to make it work as intended, so we can override it with either of the following:

```tsx
  pipeline: (defaultProcessors) => [
    ...defaultProcessors,
    {
      set: {
        field: 'entity.type',
        value: 'Identity',
      },
    },
  ],
```

or 

```tsx
  pipeline: [
    {
      set: {
        field: 'entity.type',
        value: 'Host',
      },
    },
  ],
```

romulets reviewed (changes requested):
I think we might not need the new attribute, please have a look if it makes sense!

opauloh committed:
Merge branch 'main' into entity-store/user-host-entity-type

opauloh committed:
Refactors ingest pipeline configuration

Simplifies ingest pipeline configuration by replacing
`customIngestProcessors` with a more flexible `pipeline` property.
The `pipeline` property now supports both an array of processors
and a function that can modify the default processors.

Updates entity descriptions to use the new `pipeline` property.

Removes the `customIngestProcessors` type from the entity definitions.

elasticmachine commented:
## :hourglass_flowing_sand: Build in-progress
* [Buildkite Build](https://buildkite.com/elastic/kibana-pull-request/builds/324120)
* Commit: 2d6b93e6a6e7a552b990bb7c35b3209b34515964
* Kibana Serverless Image: `docker.elastic.co/kibana-ci/kibana-serverless:pr-223498-2d6b93e6a6e7`

### History
* :broken_heart: [Build #323103](https://buildkite.com/elastic/kibana-pull-request/builds/323103) failed fe97cf0dd0e34a979efb7cfca899e3f7ce7ed65a
* :broken_heart: [Build #323091](https://buildkite.com/elastic/kibana-pull-request/builds/323091) failed 3ca3eb65b6bc8770fdb8a57f6e8da8b3786a28bf
* :broken_heart: [Build #322135](https://buildkite.com/elastic/kibana-pull-request/builds/322135) failed ed5e2facebd4b36413173cedad210bb3690bc8bc
* :broken_heart: [Build #322128](https://buildkite.com/elastic/kibana-pull-request/builds/322128) failed 7c4d41b3686ed7300dea9c7ab58ac3f80f2d3371
* :broken_heart: [Build #321633](https://buildkite.com/elastic/kibana-pull-request/builds/321633) failed f16c3d3ff580ed1c1067354f76ed6030c094597e

<!--buildkite-pr-comment-kibana-pull-request
{"builds":[{"buildStatus":{"state":"running","success":true,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/324120","number":324120,"commit":"2d6b93e6a6e7a552b990bb7c35b3209b34515964"},{"buildStatus":{"state":"failed","success":false,"hasRetries":true,"hasNonPreemptionRetries":true},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/323103","number":323103,"commit":"fe97cf0dd0e34a979efb7cfca899e3f7ce7ed65a"},{"buildStatus":{"state":"failed","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/323091","number":323091,"commit":"3ca3eb65b6bc8770fdb8a57f6e8da8b3786a28bf"},{"buildStatus":{"state":"failed","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/322135","number":322135,"commit":"ed5e2facebd4b36413173cedad210bb3690bc8bc"},{"buildStatus":{"state":"failed","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/322128","number":322128,"commit":"7c4d41b3686ed7300dea9c7ab58ac3f80f2d3371"},{"buildStatus":{"state":"failed","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/321633","number":321633,"commit":"f16c3d3ff580ed1c1067354f76ed6030c094597e"}],"number":324120}
buildkite-pr-comment-->

opauloh committed:
Increments entity store definition version

Increments the version of the entity store definitions for host and user.

This change reflects updates or modifications to the structure or behavior of these entity stores.

opauloh committed:
Simplifies ingest pipeline logic

Refactors ingest pipeline creation for entity analytics.

Streamlines the process by consolidating debug mode handling and pipeline
application. Simplifies how custom processors are added or default ones modified.

opauloh commented:
Thanks for the review @romulets, I updated it to make the `pipeline` parameter work with either function or array as it's original description states, also simplified the logic on https://github.com/opauloh/kibana/blob/8f39add67723c4ba16dc84a4310a3b4531154800/x-pack/solutions/security/plugins/security_solution/server/lib/entity_analytics/entity_store/elasticsearch_assets/ingest_pipeline.ts#L120 and added more descriptions

elasticmachine commented:
## :hourglass_flowing_sand: Build in-progress
* [Buildkite Build](https://buildkite.com/elastic/kibana-pull-request/builds/324125)
* Commit: 8f39add67723c4ba16dc84a4310a3b4531154800
* Kibana Serverless Image: `docker.elastic.co/kibana-ci/kibana-serverless:pr-223498-8f39add67723`

### Failed CI Steps
* [FTR Configs #97](https://buildkite.com/elastic/kibana-pull-request/builds/324125#0198539e-3c09-449d-b846-e0e07e7e4d24)

### Test Failures
* [[job]](https://buildkite.com/elastic/kibana-pull-request/builds/324125#0198539e-3c09-449d-b846-e0e07e7e4d24) [[logs]](https://buildkite.com/organizations/elastic/pipelines/kibana-pull-request/builds/324125/jobs/0198539e-3c09-449d-b846-e0e07e7e4d24/artifacts/019853a8-91d2-434b-8724-757a82ca2116) FTR Configs #<span></span>97 / alerting api integration security and spaces enabled - Group 2 Alerting and Actions Telemetry test telemetry should retrieve telemetry data in the expected format

### History
* :green_heart: [Build #324120](https://buildkite.com/elastic/kibana-pull-request/builds/324120) succeeded 2d6b93e6a6e7a552b990bb7c35b3209b34515964
* :broken_heart: [Build #323103](https://buildkite.com/elastic/kibana-pull-request/builds/323103) failed fe97cf0dd0e34a979efb7cfca899e3f7ce7ed65a
* :broken_heart: [Build #323091](https://buildkite.com/elastic/kibana-pull-request/builds/323091) failed 3ca3eb65b6bc8770fdb8a57f6e8da8b3786a28bf
* :broken_heart: [Build #322135](https://buildkite.com/elastic/kibana-pull-request/builds/322135) failed ed5e2facebd4b36413173cedad210bb3690bc8bc
* :broken_heart: [Build #322128](https://buildkite.com/elastic/kibana-pull-request/builds/322128) failed 7c4d41b3686ed7300dea9c7ab58ac3f80f2d3371
* :broken_heart: [Build #321633](https://buildkite.com/elastic/kibana-pull-request/builds/321633) failed f16c3d3ff580ed1c1067354f76ed6030c094597e

<!--buildkite-pr-comment-kibana-pull-request
{"builds":[{"buildStatus":{"state":"running","success":true,"hasRetries":true,"hasNonPreemptionRetries":true},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/324125","number":324125,"commit":"8f39add67723c4ba16dc84a4310a3b4531154800"},{"buildStatus":{"state":"running","success":true,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/324120","number":324120,"commit":"2d6b93e6a6e7a552b990bb7c35b3209b34515964"},{"buildStatus":{"state":"failed","success":false,"hasRetries":true,"hasNonPreemptionRetries":true},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/323103","number":323103,"commit":"fe97cf0dd0e34a979efb7cfca899e3f7ce7ed65a"},{"buildStatus":{"state":"failed","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/323091","number":323091,"commit":"3ca3eb65b6bc8770fdb8a57f6e8da8b3786a28bf"},{"buildStatus":{"state":"failed","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/322135","number":322135,"commit":"ed5e2facebd4b36413173cedad210bb3690bc8bc"},{"buildStatus":{"state":"failed","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/322128","number":322128,"commit":"7c4d41b3686ed7300dea9c7ab58ac3f80f2d3371"}],"number":324125}
buildkite-pr-comment-->

opauloh committed:
Merge branch 'main' into entity-store/user-host-entity-type

elasticmachine commented:
## :hourglass_flowing_sand: Build in-progress
* [Buildkite Build](https://buildkite.com/elastic/kibana-pull-request/builds/324560)
* Commit: 2be9fb24559884c069ec419c5a220f475b6e78a8
* Kibana Serverless Image: `docker.elastic.co/kibana-ci/kibana-serverless:pr-223498-2be9fb245598`

### History
* :broken_heart: [Build #324125](https://buildkite.com/elastic/kibana-pull-request/builds/324125) failed 8f39add67723c4ba16dc84a4310a3b4531154800
* :green_heart: [Build #324120](https://buildkite.com/elastic/kibana-pull-request/builds/324120) succeeded 2d6b93e6a6e7a552b990bb7c35b3209b34515964
* :broken_heart: [Build #323103](https://buildkite.com/elastic/kibana-pull-request/builds/323103) failed fe97cf0dd0e34a979efb7cfca899e3f7ce7ed65a
* :broken_heart: [Build #323091](https://buildkite.com/elastic/kibana-pull-request/builds/323091) failed 3ca3eb65b6bc8770fdb8a57f6e8da8b3786a28bf
* :broken_heart: [Build #322135](https://buildkite.com/elastic/kibana-pull-request/builds/322135) failed ed5e2facebd4b36413173cedad210bb3690bc8bc

<!--buildkite-pr-comment-kibana-pull-request
{"builds":[{"buildStatus":{"state":"running","success":true,"hasRetries":true,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/324560","number":324560,"commit":"2be9fb24559884c069ec419c5a220f475b6e78a8"},{"buildStatus":{"state":"failed","success":false,"hasRetries":true,"hasNonPreemptionRetries":true},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/324125","number":324125,"commit":"8f39add67723c4ba16dc84a4310a3b4531154800"},{"buildStatus":{"state":"running","success":true,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/324120","number":324120,"commit":"2d6b93e6a6e7a552b990bb7c35b3209b34515964"},{"buildStatus":{"state":"failed","success":false,"hasRetries":true,"hasNonPreemptionRetries":true},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/323103","number":323103,"commit":"fe97cf0dd0e34a979efb7cfca899e3f7ce7ed65a"},{"buildStatus":{"state":"failed","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/323091","number":323091,"commit":"3ca3eb65b6bc8770fdb8a57f6e8da8b3786a28bf"},{"buildStatus":{"state":"failed","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/322135","number":322135,"commit":"ed5e2facebd4b36413173cedad210bb3690bc8bc"}],"number":324560}
buildkite-pr-comment-->

opauloh committed:
Updates entity store asset version

Updates entity store tests to assert the correct version of enrich policy is being used.
The version has been updated to 1.1.0.

elasticmachine commented:
## :hourglass_flowing_sand: Build in-progress, with failures
* [Buildkite Build](https://buildkite.com/elastic/kibana-pull-request/builds/324580)
* Commit: 0b3bc0adf004133cf408b55c1bf1e0bb46ab24ff

### Failed CI Steps
* [FTR Configs #36](https://buildkite.com/elastic/kibana-pull-request/builds/324580#01985a48-c35d-4c69-8278-b638e07aaa56)
* [FTR Configs #36](https://buildkite.com/elastic/kibana-pull-request/builds/324580#01985a6f-17ed-45c7-96dd-098159148b8e)
* [FTR Configs #99](https://buildkite.com/elastic/kibana-pull-request/builds/324580#01985a48-c392-4c65-826e-5a59d3db5720)

### Test Failures
* [[job]](https://buildkite.com/elastic/kibana-pull-request/builds/324580#01985a48-c35d-4c69-8278-b638e07aaa56) [[logs]](https://buildkite.com/organizations/elastic/pipelines/kibana-pull-request/builds/324580/jobs/01985a48-c35d-4c69-8278-b638e07aaa56/artifacts/01985a6e-e371-4629-801c-c21d8fd1540f) FTR Configs #<span></span>36 / Core Analysis - Entity Store @<span></span>ess Entity Store Engine APIs in non-default space init should have installed the expected user resources
* [[job]](https://buildkite.com/elastic/kibana-pull-request/builds/324580#01985a6f-17ed-45c7-96dd-098159148b8e) [[logs]](https://buildkite.com/organizations/elastic/pipelines/kibana-pull-request/builds/324580/jobs/01985a6f-17ed-45c7-96dd-098159148b8e/artifacts/01985a77-9114-4951-bad1-685bde453b60) FTR Configs #<span></span>36 / Core Analysis - Entity Store @<span></span>ess Entity Store Engine APIs in non-default space init should have installed the expected user resources

### History
* :broken_heart: [Build #324560](https://buildkite.com/elastic/kibana-pull-request/builds/324560) failed 2be9fb24559884c069ec419c5a220f475b6e78a8
* :broken_heart: [Build #324125](https://buildkite.com/elastic/kibana-pull-request/builds/324125) failed 8f39add67723c4ba16dc84a4310a3b4531154800
* :green_heart: [Build #324120](https://buildkite.com/elastic/kibana-pull-request/builds/324120) succeeded 2d6b93e6a6e7a552b990bb7c35b3209b34515964
* :broken_heart: [Build #323103](https://buildkite.com/elastic/kibana-pull-request/builds/323103) failed fe97cf0dd0e34a979efb7cfca899e3f7ce7ed65a
* :broken_heart: [Build #323091](https://buildkite.com/elastic/kibana-pull-request/builds/323091) failed 3ca3eb65b6bc8770fdb8a57f6e8da8b3786a28bf

<!--buildkite-pr-comment-kibana-pull-request
{"builds":[{"buildStatus":{"state":"failing","success":false,"hasRetries":true,"hasNonPreemptionRetries":true},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/324580","number":324580,"commit":"0b3bc0adf004133cf408b55c1bf1e0bb46ab24ff"},{"buildStatus":{"state":"failed","success":false,"hasRetries":true,"hasNonPreemptionRetries":true},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/324560","number":324560,"commit":"2be9fb24559884c069ec419c5a220f475b6e78a8"},{"buildStatus":{"state":"failed","success":false,"hasRetries":true,"hasNonPreemptionRetries":true},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/324125","number":324125,"commit":"8f39add67723c4ba16dc84a4310a3b4531154800"},{"buildStatus":{"state":"running","success":true,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/324120","number":324120,"commit":"2d6b93e6a6e7a552b990bb7c35b3209b34515964"},{"buildStatus":{"state":"failed","success":false,"hasRetries":true,"hasNonPreemptionRetries":true},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/323103","number":323103,"commit":"fe97cf0dd0e34a979efb7cfca899e3f7ce7ed65a"},{"buildStatus":{"state":"failed","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/323091","number":323091,"commit":"3ca3eb65b6bc8770fdb8a57f6e8da8b3786a28bf"}],"number":324580}
buildkite-pr-comment-->

opauloh committed:
Updates entity engine asset existence check

Updates asset existence check to verify version alongside name.

This ensures that the tests are checking the correct version of the
assets are installed.

opauloh committed:
Merge branch 'main' into entity-store/user-host-entity-type

opauloh committed:
reverting version changes

reverting, as changing the version is not required at this point

opauloh committed:
Merge remote-tracking branch 'origin/entity-store/user-host-entity-type' into entity-store/user-host-entity-type

elasticmachine commented:
## :hourglass_flowing_sand: Build in-progress, with failures
* [Buildkite Build](https://buildkite.com/elastic/kibana-pull-request/builds/324904)
* Commit: a2b1f6f87d21ae196515ea12f50382b138e1064d

### Failed CI Steps
* [Build Project Image](https://buildkite.com/elastic/kibana-pull-request/builds/324904#01985d20-c9ea-4a63-9034-725463c98707)

### History
* :broken_heart: [Build #324580](https://buildkite.com/elastic/kibana-pull-request/builds/324580) failed 0b3bc0adf004133cf408b55c1bf1e0bb46ab24ff
* :broken_heart: [Build #324560](https://buildkite.com/elastic/kibana-pull-request/builds/324560) failed 2be9fb24559884c069ec419c5a220f475b6e78a8
* :broken_heart: [Build #324125](https://buildkite.com/elastic/kibana-pull-request/builds/324125) failed 8f39add67723c4ba16dc84a4310a3b4531154800
* :green_heart: [Build #324120](https://buildkite.com/elastic/kibana-pull-request/builds/324120) succeeded 2d6b93e6a6e7a552b990bb7c35b3209b34515964
* :broken_heart: [Build #323103](https://buildkite.com/elastic/kibana-pull-request/builds/323103) failed fe97cf0dd0e34a979efb7cfca899e3f7ce7ed65a

<!--buildkite-pr-comment-kibana-pull-request
{"builds":[{"buildStatus":{"state":"failing","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/324904","number":324904,"commit":"a2b1f6f87d21ae196515ea12f50382b138e1064d"},{"buildStatus":{"state":"failed","success":false,"hasRetries":true,"hasNonPreemptionRetries":true},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/324580","number":324580,"commit":"0b3bc0adf004133cf408b55c1bf1e0bb46ab24ff"},{"buildStatus":{"state":"failed","success":false,"hasRetries":true,"hasNonPreemptionRetries":true},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/324560","number":324560,"commit":"2be9fb24559884c069ec419c5a220f475b6e78a8"},{"buildStatus":{"state":"failed","success":false,"hasRetries":true,"hasNonPreemptionRetries":true},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/324125","number":324125,"commit":"8f39add67723c4ba16dc84a4310a3b4531154800"},{"buildStatus":{"state":"running","success":true,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/324120","number":324120,"commit":"2d6b93e6a6e7a552b990bb7c35b3209b34515964"},{"buildStatus":{"state":"failed","success":false,"hasRetries":true,"hasNonPreemptionRetries":true},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/323103","number":323103,"commit":"fe97cf0dd0e34a979efb7cfca899e3f7ce7ed65a"}],"number":324904}
buildkite-pr-comment-->

opauloh committed:
Merge branch 'main' into entity-store/user-host-entity-type

elasticmachine commented:
## :green_heart: Build Succeeded
* [Buildkite Build](https://buildkite.com/elastic/kibana-pull-request/builds/324944)
* Commit: ba6c5301a1cc19d83727bc14824968aee9209ffa

### Metrics [[docs](https://www.elastic.co/guide/en/kibana/master/ci-metrics.html)]
<details><summary>Unknown metric groups</summary>

#### ESLint disabled line counts
| id | [before](https://github.com/elastic/kibana/commit/b95c533bfc9077cc818f0dff722f0db5aff65d1d) | [after](https://github.com/elastic/kibana/commit/ba6c5301a1cc19d83727bc14824968aee9209ffa) | diff |
| --- | --- | --- | --- |
| `securitySolution` | 640 | 644 | +4 |

#### Total ESLint disabled count
| id | [before](https://github.com/elastic/kibana/commit/b95c533bfc9077cc818f0dff722f0db5aff65d1d) | [after](https://github.com/elastic/kibana/commit/ba6c5301a1cc19d83727bc14824968aee9209ffa) | diff |
| --- | --- | --- | --- |
| `securitySolution` | 739 | 743 | +4 |


</details>


### History
* :broken_heart: [Build #324904](https://buildkite.com/elastic/kibana-pull-request/builds/324904) failed a2b1f6f87d21ae196515ea12f50382b138e1064d
* :broken_heart: [Build #324580](https://buildkite.com/elastic/kibana-pull-request/builds/324580) failed 0b3bc0adf004133cf408b55c1bf1e0bb46ab24ff
* :broken_heart: [Build #324560](https://buildkite.com/elastic/kibana-pull-request/builds/324560) failed 2be9fb24559884c069ec419c5a220f475b6e78a8
* :broken_heart: [Build #324125](https://buildkite.com/elastic/kibana-pull-request/builds/324125) failed 8f39add67723c4ba16dc84a4310a3b4531154800
* :green_heart: [Build #324120](https://buildkite.com/elastic/kibana-pull-request/builds/324120) succeeded 2d6b93e6a6e7a552b990bb7c35b3209b34515964
* :broken_heart: [Build #323103](https://buildkite.com/elastic/kibana-pull-request/builds/323103) failed fe97cf0dd0e34a979efb7cfca899e3f7ce7ed65a

<!--buildkite-pr-comment-kibana-pull-request
{"builds":[{"buildStatus":{"state":"passed","success":true,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/324944","number":324944,"commit":"ba6c5301a1cc19d83727bc14824968aee9209ffa"},{"buildStatus":{"state":"failing","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/324904","number":324904,"commit":"a2b1f6f87d21ae196515ea12f50382b138e1064d"},{"buildStatus":{"state":"failed","success":false,"hasRetries":true,"hasNonPreemptionRetries":true},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/324580","number":324580,"commit":"0b3bc0adf004133cf408b55c1bf1e0bb46ab24ff"},{"buildStatus":{"state":"failed","success":false,"hasRetries":true,"hasNonPreemptionRetries":true},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/324560","number":324560,"commit":"2be9fb24559884c069ec419c5a220f475b6e78a8"},{"buildStatus":{"state":"failed","success":false,"hasRetries":true,"hasNonPreemptionRetries":true},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/324125","number":324125,"commit":"8f39add67723c4ba16dc84a4310a3b4531154800"},{"buildStatus":{"state":"running","success":true,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/324120","number":324120,"commit":"2d6b93e6a6e7a552b990bb7c35b3209b34515964"}],"number":324944}
buildkite-pr-comment-->

opauloh committed:
Merge branch 'main' into entity-store/user-host-entity-type

elasticmachine commented:
## :yellow_heart: Build succeeded, but was flaky
* [Buildkite Build](https://buildkite.com/elastic/kibana-pull-request/builds/326700)
* Commit: e67d53691e5cd5e6bf5b6834b7662d8909f7f615

### Failed CI Steps
* [FTR Configs #14](https://buildkite.com/elastic/kibana-pull-request/builds/326700#01987c45-7d0d-4acd-89c2-02a35c7f2780)
* [FTR Configs #23](https://buildkite.com/elastic/kibana-pull-request/builds/326700#01987c45-7d14-423f-912a-e25fac537b01)
* [FTR Configs #48](https://buildkite.com/elastic/kibana-pull-request/builds/326700#01987c45-7d26-4e7b-bd65-38b140b27036)

### Test Failures
* [[job]](https://buildkite.com/elastic/kibana-pull-request/builds/326700#01987c45-7d26-4e7b-bd65-38b140b27036) [[logs]](https://buildkite.com/organizations/elastic/pipelines/kibana-pull-request/builds/326700/jobs/01987c45-7d26-4e7b-bd65-38b140b27036/artifacts/01987c6a-cdcb-4759-9c7c-ae44d9e8389c) FTR Configs #<span></span>48 / Cloud Security Posture GET /internal/cloud_security_posture/stats KSPM Compliance Dashboard Stats API should return KSPM benchmarks V2
* [[job]](https://buildkite.com/elastic/kibana-pull-request/builds/326700#01987c45-7d14-423f-912a-e25fac537b01) [[logs]](https://buildkite.com/organizations/elastic/pipelines/kibana-pull-request/builds/326700/jobs/01987c45-7d14-423f-912a-e25fac537b01/artifacts/01987c67-1485-48f0-a240-3c6a5e4d9f7c) FTR Configs #<span></span>23 / InfraOps App Metrics UI Home page with metrics present group nodes by custom field
* [[job]](https://buildkite.com/elastic/kibana-pull-request/builds/326700#01987c45-7d0d-4acd-89c2-02a35c7f2780) [[logs]](https://buildkite.com/organizations/elastic/pipelines/kibana-pull-request/builds/326700/jobs/01987c45-7d0d-4acd-89c2-02a35c7f2780/artifacts/01987c73-f88e-4bd7-9df1-ac5a720907b6) FTR Configs #<span></span>14 / Serverless Observability feature flag testing - Deployment-agnostic API integration tests SyntheticsCustomStatusRule NumberOfChecks - Location threshold > 1 - ungrouped - 2 down locations - recoveryStrategy: firstUp should be down again

### Metrics [[docs](https://www.elastic.co/guide/en/kibana/master/ci-metrics.html)]
<details><summary>Unknown metric groups</summary>

#### ESLint disabled line counts
| id | [before](https://github.com/elastic/kibana/commit/3b13ddf8c433ff9154627c10d9faccef629ee639) | [after](https://github.com/elastic/kibana/commit/e67d53691e5cd5e6bf5b6834b7662d8909f7f615) | diff |
| --- | --- | --- | --- |
| `securitySolution` | 648 | 652 | +4 |

#### Total ESLint disabled count
| id | [before](https://github.com/elastic/kibana/commit/3b13ddf8c433ff9154627c10d9faccef629ee639) | [after](https://github.com/elastic/kibana/commit/e67d53691e5cd5e6bf5b6834b7662d8909f7f615) | diff |
| --- | --- | --- | --- |
| `securitySolution` | 747 | 751 | +4 |


</details>


### History
* :green_heart: [Build #324944](https://buildkite.com/elastic/kibana-pull-request/builds/324944) succeeded ba6c5301a1cc19d83727bc14824968aee9209ffa
* :broken_heart: [Build #324904](https://buildkite.com/elastic/kibana-pull-request/builds/324904) failed a2b1f6f87d21ae196515ea12f50382b138e1064d
* :broken_heart: [Build #324580](https://buildkite.com/elastic/kibana-pull-request/builds/324580) failed 0b3bc0adf004133cf408b55c1bf1e0bb46ab24ff
* :broken_heart: [Build #324560](https://buildkite.com/elastic/kibana-pull-request/builds/324560) failed 2be9fb24559884c069ec419c5a220f475b6e78a8
* :broken_heart: [Build #324125](https://buildkite.com/elastic/kibana-pull-request/builds/324125) failed 8f39add67723c4ba16dc84a4310a3b4531154800
* :green_heart: [Build #324120](https://buildkite.com/elastic/kibana-pull-request/builds/324120) succeeded 2d6b93e6a6e7a552b990bb7c35b3209b34515964

<!--buildkite-pr-comment-kibana-pull-request
{"builds":[{"buildStatus":{"state":"passed","success":true,"hasRetries":true,"hasNonPreemptionRetries":true},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/326700","number":326700,"commit":"e67d53691e5cd5e6bf5b6834b7662d8909f7f615"},{"buildStatus":{"state":"passed","success":true,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/324944","number":324944,"commit":"ba6c5301a1cc19d83727bc14824968aee9209ffa"},{"buildStatus":{"state":"failing","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/324904","number":324904,"commit":"a2b1f6f87d21ae196515ea12f50382b138e1064d"},{"buildStatus":{"state":"failed","success":false,"hasRetries":true,"hasNonPreemptionRetries":true},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/324580","number":324580,"commit":"0b3bc0adf004133cf408b55c1bf1e0bb46ab24ff"},{"buildStatus":{"state":"failed","success":false,"hasRetries":true,"hasNonPreemptionRetries":true},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/324560","number":324560,"commit":"2be9fb24559884c069ec419c5a220f475b6e78a8"},{"buildStatus":{"state":"failed","success":false,"hasRetries":true,"hasNonPreemptionRetries":true},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/324125","number":324125,"commit":"8f39add67723c4ba16dc84a4310a3b4531154800"}],"number":326700}
buildkite-pr-comment-->
